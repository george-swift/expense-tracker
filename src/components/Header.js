import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { FaDoorOpen } from 'react-icons/fa';
import { Link } from 'react-router-dom';
import { signOutRequest } from '../actions';
import { useVerify } from '../hooks';

const Header = ({ children }) => {
  const { loggedIn, dispatch, navigate } = useVerify();
  const handleSignOut = () => dispatch(signOutRequest());

  useEffect(() => {
    if (loggedIn === false) navigate('/', { replace: true });
  }, [loggedIn]);

  return (
    <header>
      <div className="container-fluid pe-4">
        <h2 className="theme">
          { children }
          <Link to="/" className="center" onClick={handleSignOut}>
            <span className="fs-6 me-3">Sign out</span>
            <FaDoorOpen className="fs-5" />
          </Link>
        </h2>
      </div>
    </header>
  );
};

Header.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Header;
