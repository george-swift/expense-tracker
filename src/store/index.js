import { configureStore, combineReducers } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';
import user from '../slice/user';
import lists from '../slice/lists';
import reports from '../slice/reports';
import expenses from '../slice/expenses';
import notifications from '../slice/notifications';
import rootSaga from '../sagas';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: combineReducers({
    user,
    lists,
    reports,
    expenses,
    notifications,
  }),
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().prepend(sagaMiddleware),
});

export default store;

sagaMiddleware.run(rootSaga);
