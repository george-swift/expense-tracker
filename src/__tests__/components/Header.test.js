import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../__mocks__/mockstore';
import Header from '../../components/Header';

describe('Snapshot of the Header component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Header>
              <span>Track.it</span>
            </Header>
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
