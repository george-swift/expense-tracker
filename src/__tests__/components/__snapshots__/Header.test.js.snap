// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Header component should render correctly 1`] = `
<header>
  <div
    className="container-fluid pe-4"
  >
    <h2
      className="wrap-theme"
    >
      <span>
        Track.it
      </span>
      <a
        className="center"
        href="/"
        onClick={[Function]}
      >
        <span
          className="fs-6 me-3"
        >
          Sign out
        </span>
        <svg
          className="fs-5"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 640 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M624 448h-80V113.45C544 86.19 522.47 64 496 64H384v64h96v384h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM312.24 1.01l-192 49.74C105.99 54.44 96 67.7 96 82.92V448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h336V33.18c0-21.58-19.56-37.41-39.76-32.17zM264 288c-13.25 0-24-14.33-24-32s10.75-32 24-32 24 14.33 24 32-10.75 32-24 32z"
          />
        </svg>
      </a>
    </h2>
  </div>
</header>
`;
